### PR TITLE
Donate Block: Accessibility improvements

### PR DESCRIPTION
**Pull request by laurelfulford**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR makes accessibility improvements to the Frequency layout version of the Donate block, specifically to make sure the tabs have the correct markup and attributes, and can be navigated in a predictable way via keyboard.

The tabs have been added on top of the existing radio button logic that the block used, to try to minimize the changes being made. 

See 1200550061930446-as-1203715106299306 and 1200024937525366-as-1202002498910398

### How to test the changes in this Pull Request: 

1. Apply this PR and run `npm run build`.
2. Add a donate block to the editor, and switch it to the 'Frequency' Layout. 
3. In the editor, try clicking the different tabs, and confirm that the active tab and panel updates. 
4. In the editor, try switching the 'default' tab to one that's neither the original default nor the currently selected one, and confirm that the correct tab becomes selected. Repeat step 3 to confirm the other tabs still change on click.
5. Enable Configure Manually, and try disabling one tab; confirm that the preview works, and that setting the Default tab or clicking on the tabs still works. Re-enable the tab so all are visible. 
6. Publish the block. 
7. Repeat steps 3-5 on the front-end, and confirm that the block works as expected: the tabs and content update on click, and they display as expected. 
8. Try changing the block styles to Alternate and Minimal, and confirm that things display as expected. 

Default:
image

Alternate:
image

Minimal:
image

9. It would also be good to do a visual test of the Tiered Donate block to make sure nothing has changed -- the two blocks don't share the same markup, but they do share some of the same styles, so the changes made in this PR may affect that block.

These are the specific accessibility improvements that were added in this update:

* You can now use the "tab" key to navigate between different frequencies.
* The tabs have been replaced with buttons (which is required for them to to be "tabable" with the keyboard, and is semantically correct for their function). Each of these buttons has the attribute `role="tab"`, and all three are wrapped in a parent element with the attribute `role="tablist"`.
* Each tab and its associated panel has attributes that link them together -- each tab has a unique ID and an `aria-controls` attribute that has the ID of its associated panel, and each panel has an ID and an `aria-labelledby` attribute that has the ID of its associated tab.  
* When a tab is selected, it will have the attribute `aria-selected="true"`; if not, it will have the attribute `aria-selected="false"`
* When a tab is selected, its associated panel should get the attribute `tabindex="0"`. 

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by adekbadek — CHANGES_REQUESTED** [review]
The styling of the tiers-based layout is broken – here's a before and after:

image

image

The interaction with this version of the block is broken, too:



The frequency-based version works as intended, though! Someday it'd be great to change the markup to remove the radio inputs entirely. This hacky markup was necessitated by AMP, which we don't support anymore. The block can have sane markup with JS-based interaction.

File: src/blocks/donate/frontend/class-newspack-blocks-donate-renderer-frequency-based.php [hunk 1]
```diff
@@ -40,14 +40,42 @@ private static function render_frequency_selection( $frequency_slug, $frequency_
 		/>
 		<label
 			for='newspack-donate-<?php echo esc_attr( $frequency_slug . '-' . $uid ); ?>'
-			class='wpbnbd__button freq-label'
+			class=''
```
[the remark this answers is not in the document]
Comment (laurelfulford) [reply src/blocks/donate/frontend/class-newspack-blocks-donate-renderer-frequency-based.php]: 🤦‍♀️ Yes! Not sure why I left that -- it's been removed!

File: src/blocks/donate/edit/FrequencyBasedLayout.tsx [hunk 2]
```diff
@@ -68,29 +68,52 @@ const FrequencyBasedLayout = ( props: { isTiered: boolean } & ComponentProps ) =
...
+	);
+
+	// This code is fired on tab select and updates aria elements, tabindex states, and radio buttons
 	const displayAmount = ( amount: number ) => amount.toFixed( 2 ).replace( /\.?0*$/, '' );
 
 	const renderUntieredForm = () => (
 		<div className="wp-block-newspack-blocks-donate__options">
 			<div className="wp-block-newspack-blocks-donate__frequencies frequencies">
+				<div className="tab-container">
+					{ availableFrequencies.map( frequencySlug => renderTabs( frequencySlug ) ) }
```
[the remark this answers is not in the document]
Comment (laurelfulford) [reply src/blocks/donate/edit/FrequencyBasedLayout.tsx]: I appreciate any nits! 😄

File: src/blocks/donate/edit/FrequencyBasedLayout.tsx [hunk 3]
```diff
@@ -68,29 +68,52 @@ const FrequencyBasedLayout = ( props: { isTiered: boolean } & ComponentProps ) =
 	const isRenderingStripePaymentForm =
 		window.newspack_blocks_data?.is_rendering_stripe_payment_form;
 
+	// Need to do something better with this!
```
[the remark this answers is not in the document]
Comment (laurelfulford) [reply src/blocks/donate/edit/FrequencyBasedLayout.tsx]: 🤦‍♀️ I circle backed and fixed what that comment referred to, but didn't remove the comment! It's gone now -- thanks for catching that!

**Comment by laurelfulford** [comment]
Thanks so much @adekbadek! The code updates should all be done.

> The styling of the tiers-based layout is broken – here's a before and after:

Gah, good catch! Part of this was definitely the CSS changes I added (the messed up looking tabs), and part of it was me not rebasing master and [pulling in the fix for the multiple-selected issue](https://github.com/Automattic/newspack-blocks/pull/1617) -- both should be fixed! 🙂

**Review by dkoo — APPROVED** [review]
Looking good to me! The only bug I saw is a pretty rare edge case that's only in the editor (doesn't affect the front-end at all), so let's consider it non-blocking. To replicate:

1. Set default donation settings (in Reader Revenue > Donations) to tiered and allow all three tiers.
2. Set the block to Frequency layout and note the default tab.
3. Turn on "Configure manually", then disable the frequency that was the default in step  2. Observe that the block gets put into a state with no active tab:

Screenshot 2023-12-12 at 10 12 31 AM

This limbo state is easily fixed by clicking on a tab or saving and refreshing the browser, so it's a very minor issue.

**Comment by laurelfulford** [comment]
Thanks @dkoo! 

I've spun up a separate issue to dig into that weird editor state -- that's a great catch! 

I'm going to go ahead and merge this for now, and will address that in the next sprint!

**Pull request merged**